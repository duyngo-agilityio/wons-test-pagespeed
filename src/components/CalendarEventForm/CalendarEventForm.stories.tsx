import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test';

// Components
import CalendarEventForm from './index';

const meta: Meta<typeof CalendarEventForm> = {
  title: 'Components/CalendarEventForm',
  component: CalendarEventForm,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
};

export default meta;
type Story = StoryObj<typeof CalendarEventForm>;

const mockTimeRange = {
  start: '12:00am',
  end: '12:30am',
};

export const Default: Story = {
  args: {
    eventTitle: '',
    timeRange: mockTimeRange,
    date: new Date(),
    onSubmit: fn(),
    onClose: fn(),
  },
};
