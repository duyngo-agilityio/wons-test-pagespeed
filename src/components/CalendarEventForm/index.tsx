'use client';

import { memo, useState, useCallback, useEffect } from 'react';
import { Controller, useForm } from 'react-hook-form';
import isEqual from 'react-fast-compare';
import dayjs from 'dayjs';

import { Select, SelectItem } from '@nextui-org/react';

// Constants
import { MESSAGES } from '@/constants';

// Types
import { TEventResponse } from '@/types';

// Utils
import {
  capitalizeFirstLetter,
  clearErrorOnChange,
  formatDateString,
  formatDateToISO,
  formatEventDate,
  formatToCalendarDate,
  formatToStandardDate,
  getUserIds,
  parseStringToNumberArray,
} from '@/utils';

// Api
import { getUsers } from '@/api'; // Import API to fetch users

// Models
import { IEvent, TUser } from '@/models';

// Components
import {
  ClockIcon,
  CalendarIcon,
  LocationIcon,
  PeopleIcon,
  Input,
  Button,
  Text,
  DateTimePickerModal,
  AddressInput,
} from '@/components';

interface TimeRangeProps {
  start: string;
  end: string;
}

interface EventForm {
  title: string;
  location?: string;
  people?: string | number[];
}

interface CalendarEventFormProps {
  title: string;
  previewData?: EventForm | null;
  timeRange: TimeRangeProps;
  eventTitle: string;
  date: Date;
  user: TUser;
  repeatSetting?: string;
  onSubmit: (data: Partial<IEvent>) => void;
  onClose: () => void;
}

const CalendarEventForm = ({
  previewData,
  title = '',
  timeRange,
  eventTitle,
  date,
  user,
  repeatSetting = 'does not repeat',
  onClose,
  onSubmit,
}: CalendarEventFormProps) => {
  const [isUserListOpen, setIsUserListOpen] = useState(false);
  const [users, setUsers] = useState<TUser[]>([]); // State to store user list
  const [isDateTimePickerOpen, setIsDateTimePickerOpen] = useState(false);
  const [calendarDate, setCalendarDate] = useState(formatToCalendarDate(date));
  const [startTime, setStartTime] = useState(timeRange.start);
  const [endTime, setEndTime] = useState(timeRange.end);

  const {
    handleSubmit,
    control,
    clearErrors,
    reset,
    formState: { errors, isValid },
  } = useForm<EventForm>({
    mode: 'onBlur',
    reValidateMode: 'onBlur',
    defaultValues: previewData
      ? {
          title: previewData?.title || '',
          location: previewData?.location,
          people: getUserIds(previewData as TEventResponse) || '',
        }
      : {
          title: eventTitle,
          location: '',
          people: '',
        },
  });

  const standardDate = formatToStandardDate(calendarDate);
  const formattedDate = formatEventDate(standardDate);

  const toggleDateTimePicker = useCallback(() => {
    setIsDateTimePickerOpen((prev) => !prev);
  }, [setIsDateTimePickerOpen]);

  const handleFormSubmit = handleSubmit((data) => {
    const people = parseStringToNumberArray(data.people as string);

    const formattedStart = formatDateToISO(
      new Date(formatDateString(calendarDate)),
      dayjs(startTime, 'hh:mma').utc(true).format('HH:mm'),
    );
    const formattedEnd = formatDateToISO(
      new Date(formatDateString(calendarDate)),
      dayjs(endTime, 'hh:mma').utc(true).format('HH:mm'),
    );

    onSubmit({
      ...data,
      users_permissions_users: [Number(user.id), ...people],
      startTime: formattedStart,
      endTime: formattedEnd,
      date: new Date(formatDateString(calendarDate)),
    });
  });

  const handleModalClose = () => {
    reset({ title });
    onClose();
  };

  const toggleUserList = useCallback(async () => {
    const response = await getUsers(); // Call API to fetch users

    if (response) {
      setUsers(response);
    } else {
      console.error('No data found in the response');
    }
  }, []);

  useEffect(() => {
    toggleUserList(); // Automatically fetch users when component mounts
  }, [toggleUserList]);

  const [isOpenLocation, setIsOpenLocation] = useState(false);

  const usersOptions = users
    .filter((u) => u.id.toString() !== user.id.toString())
    .map((user) => ({
      label: user.username,
      key: user.id,
    }));

  return (
    <>
      <form onSubmit={handleFormSubmit}>
        <Controller
          name="title"
          control={control}
          rules={{
            required: MESSAGES.ERROR.FIELD_REQUIRED,
          }}
          render={({
            field: { name, onChange, ...rest },
            fieldState: { error },
          }) => (
            <Input
              classNames={{
                inputWrapper: 'w-full h-[42px] rounded-[10px] bg-gray-200/30',
                input:
                  'placeholder:text-blue-800 placeholder:opacity-20 placeholder:font-medium placeholder:text-[16px]',
              }}
              type="text"
              isInvalid={!!error}
              placeholder="Add title"
              errorMessage={error?.message}
              onChange={(e) => {
                onChange(e.target.value);
                clearErrorOnChange(name, errors, clearErrors);
              }}
              {...rest}
            />
          )}
        />

        <div
          className="flex mt-[25px] gap-[0_15px]"
          onClick={toggleDateTimePicker}
        >
          <Button
            className="!bg-pink-50 dark:!bg-pink-600 text-pink-500 dark:text-pink-500 border-none rounded-full w-10 h-10 flex justify-center items-center cursor-pointer !px-0"
            data-testid="time-button"
            onClick={toggleDateTimePicker}
          >
            <ClockIcon />
          </Button>

          <div>
            <div className="flex gap-1">
              <Text
                className="text-blue-800 text-[12px] font-normal leading-normal col-span-1"
                text={`${formattedDate}`} // Display date
              />
              <Text as="span" text="-" />
              <Text
                className="text-blue-800 text-[12px] font-normal leading-normal col-span-1 uppercase"
                text={`${startTime} - ${endTime}`} // Display time
              />
            </div>

            <Text
              className="text-[rgba(1, 13, 28, 0.50)] text-opacity-50 text-[12px] font-normal leading-normal col-span-2"
              text={`Time zone - ${capitalizeFirstLetter(repeatSetting)}`}
            />
          </div>
        </div>

        {(isUserListOpen || previewData) && (
          <Controller
            name="people"
            control={control}
            render={({ field: { onChange, value, name, ...rest } }) => (
              <Select
                selectionMode="multiple"
                label="Add People"
                defaultSelectedKeys={value}
                placeholder=" "
                labelPlacement="outside"
                variant="flat"
                classNames={{
                  trigger:
                    'w-full bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus:bg-gray-50 dark:focus:bg-gray-600 py-[26px] mt-5',
                  label: 'text-xl font-medium pb-1',
                }}
                onChange={(e) => {
                  onChange(e.target.value);
                  clearErrorOnChange(name, errors, clearErrors);
                }}
                {...rest}
              >
                {usersOptions.map((option) => (
                  <SelectItem key={option.key}>{option.label}</SelectItem>
                ))}
              </Select>
            )}
          />
        )}

        {(isOpenLocation || previewData?.location) && (
          <Controller
            name="location"
            control={control}
            render={({
              field: { name, onChange, ...rest },
              fieldState: { error },
            }) => (
              <AddressInput
                isInvalid={!!error}
                errorMessage={error?.message}
                label="Location"
                classNames={{ mainWrapper: 'mt-5' }}
                onChange={(value) => {
                  onChange(value);

                  // Clear error message on change
                  clearErrorOnChange(name, errors, clearErrors);
                }}
                {...rest}
              />
            )}
          />
        )}

        <div className="flex gap-[0_25px] m-[30px_0]">
          {!isUserListOpen && !previewData && (
            <Button
              color="primary"
              startContent={<PeopleIcon />}
              className="text-[15px] font-medium w-auto  py-[10px] px-[25px] mt-10 mt-0"
              onClick={() => setIsUserListOpen((prev) => !prev)}
            >
              Add People
            </Button>
          )}

          {!isOpenLocation && !previewData?.location && (
            <Button
              startContent={<LocationIcon />}
              className="!bg-white font-medium dark:!bg-white text-center !text-blue-500 dark:text-white/70 border border-[1px] border-[rgba(58, 54, 219, 0.1)] py-[10px] px-[25px] !rounded-[10px] font-DM-Sans text-[14.22px] font-normal leading-normal"
              onClick={() => setIsOpenLocation(true)}
            >
              Add Location
            </Button>
          )}
        </div>

        <div className="m-[0_0_40px] flex gap-[0_15px]">
          <Button
            className="!bg-pink-50 dark:!bg-pink-600 text-pink-500 dark:text-pink-500 border-none rounded-full w-10 h-10 flex justify-center items-center cursor-pointer !px-0"
            data-testid="calendar-button"
          >
            <CalendarIcon width={15} height={15} color="pink-500" />
          </Button>

          <div className="flex flex-col gap-[0_30px]">
            <Text
              className="text-blue-800 text-[12px] font-normal leading-normal col-span-1"
              text={user.fullName} // Display name
            />
            <Text
              className="text-[rgba(1, 13, 28, 0.50)] text-opacity-50 text-[12px] font-normal leading-normal"
              text="Busy - Default visibility - notify 30 minutes before" // Display status
            />
          </div>
        </div>

        <div className="flex flex-row-reverse gap-[0_20px]">
          <Button
            className="min-w-[93px] text-[15px] font-normal w-auto py-[10px] mt-10 mt-0"
            type="submit"
            color="primary"
            isDisabled={!isValid}
          >
            Save
          </Button>

          <Button
            className="min-w-[93px] !bg-white font-normal dark:!bg-white text-center !text-blue-500 dark:text-white/70 border border-[1px] border-[rgba(58, 54, 219, 0.1)] py-[10px] !rounded-[10px] font-DM-Sans text-[15px] font-normal leading-normal"
            onClick={handleModalClose}
          >
            Close
          </Button>
        </div>
      </form>

      {/* DateTimePickerModal */}
      {isDateTimePickerOpen && (
        <DateTimePickerModal
          isOpen={isDateTimePickerOpen}
          onClose={toggleDateTimePicker}
          selectedDate={formatDateString(calendarDate)}
          selectedStartTime={startTime}
          selectedEndTime={endTime}
          onDateChange={(dateString) =>
            setCalendarDate(formatToCalendarDate(new Date(dateString)))
          }
          onStartTimeChange={setStartTime}
          onEndTimeChange={setEndTime}
        />
      )}
    </>
  );
};

export default memo(CalendarEventForm, isEqual);
