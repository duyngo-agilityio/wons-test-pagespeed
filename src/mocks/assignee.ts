import { TRole } from '@/types';

export const MOCK_ASSIGNEES = {
  data: [
    {
      id: 1,
      attributes: {
        name: 'John Doe',
        avatar: '/avatar1.jpg',
        email: 'john@example.com',
        username: 'john_doe',
        fullName: 'John Doe',
        password: 'password123',
        role: 'User' as unknown as TRole,
        token: 'token123',
      },
    },
    {
      id: 2,
      attributes: {
        name: 'Jane Smith',
        avatar: '/avatar2.jpg',
        email: 'jane@example.com',
        username: 'jane_smith',
        fullName: 'Jane Smith',
        password: 'password123',
        role: 'User' as unknown as TRole,
        token: 'token123',
      },
    },
  ],
};
