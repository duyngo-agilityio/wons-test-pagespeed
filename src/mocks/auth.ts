// Types
import { ISignUpFormData } from '@/types';

export const SIGN_UP_FORM_DATA_MOCK: ISignUpFormData = {
  fullName: 'John Doe',
  username: 'johndoe',
  email: 'john.doe@example.com',
  password: 'Abcd@0987',
};

export const MOCK_SUCCESS_SIGN_UP_RES = {
  data: { user: { ...SIGN_UP_FORM_DATA_MOCK, id: '1' } },
  totalCount: 0,
};
