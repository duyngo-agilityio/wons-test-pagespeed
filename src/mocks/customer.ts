// Models
import { ICustomer } from '@/models';

// Types
import { TableColumnType } from '@/types';

export const PRODUCT_TABLE_COLUMNS_MOCK: TableColumnType<Partial<ICustomer>>[] =
  [
    {
      header: 'Name',
      accessor: 'name',
      isSort: true,
    },
    {
      header: 'FirstName',
      accessor: 'firstName',
      isSort: true,
    },
    {
      header: 'LastName',
      accessor: 'lastName',
      isSort: true,
    },
    {
      header: 'Email',
      accessor: 'email',
      isSort: true,
    },
  ];

export const PRODUCT_MOCK = [
  {
    id: '1',
    name: 'John Doe',
    firstName: 'John',
    lastName: 'Doe',
    email: 'john.doe@example.com',
    phone: '123-456-7890',
    job: 'Software Engineer',
    address: '123 Main St, City, Country',
    avatar: 'https://example.com/avatar1.jpg',
  },
  {
    id: '2',
    name: 'Jane Smith',
    firstName: 'Jane',
    lastName: 'Smith',
    email: 'jane.smith@example.com',
    phone: '098-765-4321',
    job: 'Project Manager',
    address: '456 Elm St, City, Country',
    avatar: 'https://example.com/avatar2.jpg',
  },
  {
    id: '3',
    name: 'Alice Johnson',
    firstName: 'Alice',
    lastName: 'Johnson',
    email: 'alice.johnson@example.com',
    phone: '555-666-7777',
    job: 'UX Designer',
    address: '789 Oak St, City, Country',
    avatar: 'https://example.com/avatar3.jpg',
  },
  {
    id: '4',
    name: 'Bob Williams',
    firstName: 'Bob',
    lastName: 'Williams',
    email: 'bob.williams@example.com',
    phone: '111-222-3333',
    job: 'Data Scientist',
    address: '101 Pine St, City, Country',
    avatar: 'https://example.com/avatar4.jpg',
  },
  {
    id: '5',
    name: 'Charlie Brown',
    firstName: 'Charlie',
    lastName: 'Brown',
    email: 'charlie.brown@example.com',
    phone: '444-555-6666',
    job: 'Marketing Specialist',
    address: '202 Birch St, City, Country',
    avatar: 'https://example.com/avatar5.jpg',
  },
];
