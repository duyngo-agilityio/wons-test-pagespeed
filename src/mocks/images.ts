export const MOCK_IMAGES = [
  'https://media.wired.com/photos/6500ad57fe61eb702d721b58/1:1/w_1800,h_1800,c_limit/Apple-iPhone-15-Pro-Hero-Gear.jpg',
  'https://clickbuy.com.vn/uploads/images/2019/09/thumb_iP11_3.jpg',
];
