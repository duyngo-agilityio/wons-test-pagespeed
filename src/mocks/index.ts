export * from './customer';
export * from './recentServices';
export * from './tabs';
