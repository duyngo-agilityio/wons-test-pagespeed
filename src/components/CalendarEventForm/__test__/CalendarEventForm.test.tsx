import { render } from '@testing-library/react';

// Mocks
import { MOCK_USERS } from '@/mocks';

// Components
import CalendarEventForm from '..';

describe('CalendarEventForm', () => {
  const mockDate = new Date(2024, 1, 1);
  const mockTimeRange = { start: '05:00am', end: '07:00am' };
  const onCloseMock = jest.fn();
  const onSubmitMock = jest.fn();

  it('render successful', () => {
    const { container } = render(
      <CalendarEventForm
        eventTitle="example title"
        date={mockDate}
        timeRange={mockTimeRange}
        user={MOCK_USERS[0]}
        previewData={null}
        onClose={onCloseMock}
        onSubmit={onSubmitMock}
        repeatSetting="does not repeat"
      />,
    );

    // Check if the container is in the document
    expect(container).toBeInTheDocument();

    // Match the snapshot
    expect(container).toMatchSnapshot();
  });
});
