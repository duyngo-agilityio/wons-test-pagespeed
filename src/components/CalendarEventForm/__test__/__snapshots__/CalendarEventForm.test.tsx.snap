// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarEventForm render successful 1`] = `
<div>
  <form>
    <div
      class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[29px] is-filled"
      data-filled="true"
      data-filled-within="true"
      data-has-value="true"
      data-slot="base"
    >
      <div
        class="h-full flex flex-col"
        data-slot="main-wrapper"
      >
        <div
          class="relative inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary group-data-[focus=true]:bg-primary-50 placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus-within:!bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600 p-3.75 w-full h-[42px] rounded-[10px] bg-gray-200/30 is-filled"
          data-slot="input-wrapper"
          style="cursor: text;"
        >
          <div
            class="inline-flex w-full items-center h-full box-border"
            data-slot="inner-wrapper"
          >
            <input
              aria-label="Add title"
              class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !text-blue-800/70 dark:!text-white/70 dark:placeholder:text-white/40 bg-transparent text-md placeholder:text-blue-800 placeholder:opacity-20 placeholder:font-medium placeholder:text-[16px] is-filled"
              data-filled="true"
              data-filled-within="true"
              data-slot="input"
              id="react-aria-:r0:"
              placeholder="Add title"
              type="text"
              value="example title"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex mt-[25px] gap-[0_15px]"
    >
      <button
        class="z-0 group relative box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default data-[hover=true]:opacity-hover border-transparent hover:!bg-gray-200/50 dark:hover:!bg-gray-900 px-5 pb-2.5 pt-[11px] text-md border-[1px] min-w-max !bg-pink-50 dark:!bg-pink-600 text-pink-500 dark:text-pink-500 border-none rounded-full w-10 h-10 flex justify-center items-center cursor-pointer !px-0"
        data-testid="time-button"
        type="button"
      >
        <div
          class="flex items-center"
        >
          <svg
            color="#FF69B4"
            fill="none"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M8 16C3.584 16 0 12.424 0 8C0 3.584 3.584 0 8 0C12.424 0 16 3.584 16 8C16 12.424 12.424 16 8 16ZM10.5523 10.9681C10.6483 11.0241 10.7523 11.0561 10.8643 11.0561C11.0643 11.0561 11.2643 10.9521 11.3763 10.7601C11.5443 10.4801 11.4563 10.1121 11.1683 9.93614L8.32031 8.24014V4.54414C8.32031 4.20814 8.04831 3.94414 7.72031 3.94414C7.39231 3.94414 7.12031 4.20814 7.12031 4.54414V8.58414C7.12031 8.79214 7.23231 8.98414 7.41631 9.09614L10.5523 10.9681Z"
              fill="#FF69B4"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
      <div>
        <div
          class="flex gap-1"
        >
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-md text-blue-800 text-[12px] font-normal leading-normal col-span-1"
          >
            Thursday, February 1, 2024
          </p>
          <span
            class="font-dm-sans text-blue-800 dark:text-white text-md "
          >
            -
          </span>
          <p
            class="font-dm-sans text-blue-800 dark:text-white text-md text-blue-800 text-[12px] font-normal leading-normal col-span-1 uppercase"
          >
            05:00am - 07:00am
          </p>
        </div>
        <p
          class="font-dm-sans text-blue-800 dark:text-white text-md text-[rgba(1, 13, 28, 0.50)] text-opacity-50 text-[12px] font-normal leading-normal col-span-2"
        >
          Time zone - Does not repeat
        </p>
      </div>
    </div>
    <div
      class="flex gap-[0_25px] m-[30px_0]"
    >
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-blue-500 border-transparent dark:bg-purple-600 text-white hover:bg-blue-100 rounded-[10px] border-[1px] h-auto min-w-max text-[15px] font-medium w-auto py-[10px] px-[25px] mt-0"
        type="button"
      >
        <svg
          aria-hidden="true"
          color="#FFFFFF"
          fill="none"
          focusable="false"
          height="13"
          tabindex="-1"
          viewBox="0 0 18 13"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.4019 3.43921C12.4019 5.34833 10.8693 6.87915 8.9579 6.87915C7.04654 6.87915 5.51394 5.34833 5.51394 3.43921C5.51394 1.52935 7.04654 0 8.9579 0C10.8693 0 12.4019 1.52935 12.4019 3.43921ZM8.95831 12.9998C6.15098 12.9998 3.75391 12.5578 3.75391 10.79C3.75391 9.02132 6.1356 8.56325 8.95831 8.56325C11.7656 8.56325 14.1627 9.00522 14.1627 10.7739C14.1627 12.5418 11.781 12.9998 8.95831 12.9998ZM13.874 3.50139C13.874 4.47461 13.5817 5.38125 13.0689 6.13494C13.0161 6.21251 13.063 6.31715 13.156 6.33325C13.2843 6.3552 13.4169 6.36764 13.5517 6.3713C14.896 6.40642 16.1026 5.54223 16.4359 4.24118C16.9297 2.30864 15.4799 0.573672 13.6337 0.573672C13.433 0.573672 13.241 0.594893 13.0542 0.632944C13.0286 0.638798 13.0015 0.650506 12.9868 0.67319C12.9692 0.700996 12.9824 0.738315 13 0.762463C13.5546 1.53885 13.874 2.48573 13.874 3.50139ZM16.1003 7.72879C17.0036 7.90514 17.5977 8.26516 17.8439 8.78836C18.0519 9.2179 18.0519 9.71622 17.8439 10.145C17.4673 10.9565 16.2534 11.217 15.7816 11.2843C15.6842 11.299 15.6058 11.2148 15.616 11.1175C15.8571 8.86885 13.9398 7.8027 13.4439 7.55756C13.4226 7.54659 13.4182 7.52976 13.4204 7.51951C13.4219 7.5122 13.4307 7.50049 13.4468 7.49829C14.5201 7.47853 15.6739 7.62488 16.1003 7.72879ZM4.44785 6.37081C4.58265 6.36715 4.71452 6.35545 4.84346 6.33276C4.9365 6.31666 4.98339 6.21202 4.93064 6.13446C4.41782 5.38076 4.12551 4.47413 4.12551 3.5009C4.12551 2.48524 4.44492 1.53836 4.9995 0.761978C5.01709 0.73783 5.02954 0.700511 5.01269 0.672705C4.99804 0.650752 4.9702 0.638313 4.94529 0.632459C4.75775 0.594408 4.5658 0.573187 4.36507 0.573187C2.51891 0.573187 1.06909 2.30816 1.5636 4.2407C1.89693 5.54174 3.10353 6.40594 4.44785 6.37081ZM4.57931 7.51927C4.58151 7.53024 4.57712 7.54634 4.5566 7.55805C4.0599 7.80318 2.14268 8.86934 2.3837 11.1173C2.39396 11.2153 2.31631 11.2987 2.21887 11.2848C1.74707 11.2175 0.533151 10.957 0.156594 10.1455C-0.0521979 9.71597 -0.0521979 9.21838 0.156594 8.78885C0.402748 8.26565 0.996156 7.90563 1.89945 7.72855C2.32656 7.62537 3.47968 7.47902 4.55367 7.49878C4.56979 7.50097 4.57785 7.51268 4.57931 7.51927Z"
            fill="#FFFFFF"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="flex items-center"
        >
          Add People
        </div>
      </button>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default data-[hover=true]:opacity-hover text-blue-800 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 rounded-[10px] h-auto min-w-max !bg-white dark:!bg-white text-center !text-blue-500 dark:text-white/70 border-[1px] border-[rgba(58, 54, 219, 0.1)] py-[10px] px-[25px] !rounded-[10px] font-DM-Sans text-[14.22px] font-normal leading-normal"
        type="button"
      >
        <svg
          aria-hidden="true"
          color="#3A36DB"
          fill="none"
          focusable="false"
          height="12"
          tabindex="-1"
          viewBox="0 0 11 12"
          width="11"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0 4.99065C0 2.23073 2.30633 0 5.09607 0C7.89367 0 10.2 2.23073 10.2 4.99065C10.2 6.38141 9.6942 7.67258 8.8617 8.76695C7.94328 9.97411 6.81129 11.0259 5.53713 11.8515C5.24551 12.0423 4.98232 12.0567 4.66227 11.8515C3.38084 11.0259 2.24885 9.97411 1.3383 8.76695C0.50519 7.67258 0 6.38141 0 4.99065ZM3.41621 5.14606C3.41621 6.07064 4.17067 6.79782 5.09574 6.79782C6.02142 6.79782 6.78314 6.07064 6.78314 5.14606C6.78314 4.22869 6.02142 3.46611 5.09574 3.46611C4.17067 3.46611 3.41621 4.22869 3.41621 5.14606Z"
            fill="#3A36DB"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="flex items-center"
        >
          Add Location
        </div>
      </button>
    </div>
    <div
      class="m-[0_0_40px] flex gap-[0_15px]"
    >
      <button
        class="z-0 group relative box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default data-[hover=true]:opacity-hover border-transparent hover:!bg-gray-200/50 dark:hover:!bg-gray-900 px-5 pb-2.5 pt-[11px] text-md border-[1px] min-w-max !bg-pink-50 dark:!bg-pink-600 text-pink-500 dark:text-pink-500 border-none rounded-full w-10 h-10 flex justify-center items-center cursor-pointer !px-0"
        data-testid="calendar-button"
        type="button"
      >
        <div
          class="flex items-center"
        >
          <svg
            color="pink-500"
            height="15"
            viewBox="0 0 18 20"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="m13.41.769.002.75c2.754.215 4.574 2.092 4.577 4.97L18 14.917c.004 3.138-1.968 5.069-5.128 5.074l-7.72.01c-3.14.004-5.137-1.973-5.141-5.12L0 6.553c-.004-2.898 1.752-4.77 4.506-5.023V.78a.746.746 0 0 1 .758-.77.747.747 0 0 1 .762.769v.7l5.865-.009v-.7A.747.747 0 0 1 12.65 0a.753.753 0 0 1 .76.769ZM1.522 6.862l14.949-.02v-.35c-.043-2.15-1.121-3.277-3.056-3.445v.77c0 .43-.334.77-.758.77a.752.752 0 0 1-.762-.768v-.81l-5.865.008v.81c0 .43-.324.77-.759.77a.752.752 0 0 1-.76-.769l-.002-.77c-1.925.193-2.99 1.325-2.987 3.493v.31Zm10.719 4.542v.011c.01.46.385.809.84.799a.823.823 0 0 0 .789-.852.83.83 0 0 0-.82-.797.828.828 0 0 0-.81.84Zm.815 4.488a.848.848 0 0 1-.821-.848.833.833 0 0 1 .809-.852h.01c.463 0 .84.379.84.848a.844.844 0 0 1-.838.852ZM8.172 11.42c.02.46.396.819.85.799a.823.823 0 0 0 .78-.86.816.816 0 0 0-.821-.8c-.454.02-.81.402-.809.861Zm.854 4.427a.823.823 0 0 1-.85-.798c0-.46.355-.84.809-.861a.815.815 0 0 1 .82.797c.02.461-.335.841-.779.862Zm-4.922-4.392c.02.46.396.82.85.799.445-.02.8-.4.778-.86a.815.815 0 0 0-.82-.8.852.852 0 0 0-.808.861Zm.854 4.397a.822.822 0 0 1-.85-.799.853.853 0 0 1 .809-.86.815.815 0 0 1 .82.799c.021.46-.333.84-.779.86Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
      <div
        class="flex flex-col gap-[0_30px]"
      >
        <p
          class="font-dm-sans text-blue-800 dark:text-white text-md text-blue-800 text-[12px] font-normal leading-normal col-span-1"
        >
          John Doe
        </p>
        <p
          class="font-dm-sans text-blue-800 dark:text-white text-md text-[rgba(1, 13, 28, 0.50)] text-opacity-50 text-[12px] font-normal leading-normal"
        >
          Busy - Default visibility - notify 30 minutes before
        </p>
      </div>
    </div>
    <div
      class="flex flex-row-reverse gap-[0_20px]"
    >
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-blue-500 border-transparent dark:bg-purple-600 text-white hover:bg-blue-100 opacity-50 cursor-not-allowed px-5 rounded-[10px] border-[1px] h-auto min-w-[93px] text-[15px] font-normal w-auto py-[10px] mt-0"
        data-disabled="true"
        disabled=""
        type="submit"
      >
        <div
          class="flex items-center"
        >
          Save
        </div>
      </button>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default data-[hover=true]:opacity-hover text-blue-800 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 px-5 rounded-[10px] h-auto min-w-[93px] !bg-white dark:!bg-white text-center !text-blue-500 dark:text-white/70 border-[1px] border-[rgba(58, 54, 219, 0.1)] py-[10px] !rounded-[10px] font-DM-Sans text-[15px] font-normal leading-normal"
        type="button"
      >
        <div
          class="flex items-center"
        >
          Close
        </div>
      </button>
    </div>
  </form>
</div>
`;
